[Li-Zizhen] iP

Li-Zizhen opened the pull request:
# Zhen   

Zhen helps you to remember things you need to do. It is:
- Text-based
- Easy to learn
- Fast to use
- **Totoally free**

You need:
1. download it from [here](https://nus-cs2103-ay2324s2.github.io/website/schedule/week4/project.html)
2. double-click it.
3. add your task following the guide
And Zhen will help you manage the rest

Features:
- [ ] managing tasks
- [ ] managing deadlines
- [ ] find your tasks


This page's content is partially adopted from [here](https://nus-cs2103-ay2324s2.github.io/website/schedule/week4/project.html)

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

Li-Zizhen committed:
solve: level0

Li-Zizhen committed:
solve: level1

Li-Zizhen committed:
solved: level2

Li-Zizhen committed:
solved: level3

Li-Zizhen committed:
solved: level4

Li-Zizhen committed:
solved: A-TextUiTesting

Li-Zizhen committed:
solved: level5

Li-Zizhen committed:
sovled: level6

Li-Zizhen committed:
Solved: Level7

Li-Zizhen committed:
Solved: Level8

Li-Zizhen committed:
Merge branch 'branch-Level-8'

Li-Zizhen committed:
Draft for more oop

Li-Zizhen committed:
Solved: A-MoreOOP

Li-Zizhen committed:
solved: A-Packages

Li-Zizhen committed:
Merge remote-tracking branch 'origin/add-gradle-support'

Li-Zizhen committed:
Solved: A-Gradle

Li-Zizhen committed:
Solved: A-JUnit

Li-Zizhen committed:
Solved: A-JUnit

bertrandong commented on src/main/java/Duke/command/Command.java:
```diff
@@ -0,0 +1,8 @@
+import Duke.*;
+import Duke.task.*;
+public abstract class Command{
+    public abstract void execute(TaskList tsk, Ui ui, Storage storage);
```
Perhaps you could standardise your naming of tsk, with 's' or no 's' throughout all your overridden methods, to help with readability

bertrandong commented on src/main/java/Duke/task/task.java:
```diff
@@ -0,0 +1,36 @@
+        return message;
+    }
+    @Override
+    public String toString(){
```
Try to add a white space between method name and curly brazes to improve readability.

bertrandong commented on src/main/java/Duke/task/Deadline.java:
```diff
@@ -0,0 +1,33 @@
+
+public class Deadline extends task {
+    LocalDate date = null;
+    String date1;
```
Consider renaming this variable to toDate for easier readability.

bertrandong commented on src/main/java/Duke/Duke.java:
```diff
@@ -0,0 +1,33 @@
+package Duke;
+import Duke.command.*;
+import Duke.task.*;
```
Remember to import your classes one by one.

bertrandong reviewed (commented):
Overall, I found your code easy to understand. Just noted a few minor formatting and naming coding standard violations for your consideration!

Li-Zizhen committed:
Solved:Level-9: Find

Li-Zizhen committed:
Solved: A-JavaDoc

Li-Zizhen committed:
Solved: A-CodingStandard

Li-Zizhen committed:
Merge branch 'Level-9' to master

# Conflicts:
#	src/main/java/Duke/Storage.java

Li-Zizhen committed:
Merge branch 'A-CodingStandard'

Li-Zizhen committed:
Config: checkstyle

Li-Zizhen committed:
Solved: A-CheckStyle

Li-Zizhen committed:
Merge branch 'A-CheckStyle'

Li-Zizhen committed:
Solved: Level-10

Li-Zizhen committed:
Merge branch 'branch-Level-10'

Li-Zizhen committed:
no message

Li-Zizhen committed:
Add assertion: Add assertion to invariants.

Li-Zizhen committed:
Merge pull request #2 from Li-Zizhen/branch-A-Assertions

Add assertion: Add assertion to invariants.

Li-Zizhen committed:
Merge branch 'master' of https://github.com/Li-Zizhen/ip

Li-Zizhen committed:
Rename package: rename Duke package to zhen.

Java coding standard says package names should be in lowercase.

Let's rename the package name from Duke to zhen.

Changing from Duke to zhen,
* follows coding standard
* shows name of my program

Li-Zizhen committed:
Parser class: split parser method to smaller methods.

Need to avoid long methods.

Split parser methods into parser specific for particular command
e.g. pareAddCommand, parseMarkingCommands, etc.

This split can organize similar logic together, makes the code more
readable.

Li-Zizhen committed:
Merge pull request #3 from Li-Zizhen/branch-A-CodeQuality

Branch a code quality

Li-Zizhen committed:
Merge branch 'master' of https://github.com/Li-Zizhen/ip

Li-Zizhen committed:
change a field's name from duke to zhen

Li-Zizhen committed:
solved: adding tag

Li-Zizhen committed:
Bug fix: resolve index out of bound error in TaskList's toString

Previous implementation will throw error when listing the taskList
if the taskList is empty.

Let's add condition to check if list is empty or not before
accessing the last task in the list, and inform user the list is
currently empty if it is.

This change will resolve the error and inform user to add tasks.

Li-Zizhen committed:
Finishes UserGuide, and fixes some bugs.

Li-Zizhen committed:
Fixes minor grammar mistakes

Li-Zizhen committed:
minor bug fixing

Li-Zizhen committed:
Fixes an GUI bug.

Li-Zizhen committed:
Add more testcases

Li-Zizhen committed:
Adds more testcases

Li-Zizhen committed:
Fixes coding standard related issues.

Li-Zizhen committed:
Fixes a minor mistake in README.md

Li-Zizhen committed:
Fixes minor naming mistake
